python/ray/data/_internal/logical/interfaces.py

```diff
@@ -0,0 +1,89 @@
+from typing import List
+
+
+class Operator:
+    """Abstract class for operators.
+
+    Operators are stateful and non-serializable; they live on the driver side of the
```

Comment:
Btw could you add a dummy `__reduce__` to enforce non-serializability?

Reply:
@ericl - actually we have to allow `Operator` and `LogicalOperator` to be serializable, because `Dataset` needs to have logical operators, and `Dataset` is expected to be serializable from training, o.w. it triggers CI test failure like [this](https://gist.github.com/c21/bd0f3376fda9ddd24e2c16d10b19c3c9).

So changed to only make `PhysicalOperator` undeserializable now. Let me know if it makes sense or not.